Create check_tweets.py

arbolitoloco1 opened the pull request.

arbolitoloco1 committed:
Create check_tweets.py

arbolitoloco1 committed:
Update check_tweets.py

arbolitoloco1 committed:
Update check_tweets.py

arbolitoloco1 committed:
Update check_tweets.py

arbolitoloco1 committed:
Update check_tweets.py

arbolitoloco1 committed:
Update check_tweets.py

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,54 @@
+import mwclient
```
you can use `mwrogue` library which is designed to support the esports wikis, and then:

```
from mwrogue.esports_client import EsportsClient
from mwrogue.esports_client import AuthCredentials
```

the 2nd line will allow you to log in, so that you can report errors on the wiki instead of printing them

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,54 @@
+def main():
+	client = tweepy.Client(bearer_token=bearertoken)
+
+	site = mwclient.Site('lol.fandom.com', path='/')
```
```
    credentials = AuthCredentials(user_file='me')
    site = EsportsClient('lol', credentials=credentials)
```

the library will help prompt you to set up a login session file locally with your bot password (you can set this up at `Special:BotPasswords` on the wiki)

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,54 @@
+checkfrom = date.today() - timedelta(days=4)
+
+def main():
+	client = tweepy.Client(bearer_token=bearertoken)
```
maybe call this `twitter_client` to avoid confusion with `site.client` as the wiki client

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,54 @@
+	client = tweepy.Client(bearer_token=bearertoken)
+
+	site = mwclient.Site('lol.fandom.com', path='/')
+	response = site.api('cargoquery',
```
there's a built-in `site.cargo_client.query()` method in `mwrogue`

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,54 @@
+	print("I have the data from the wiki!")
+
+	for source in response["cargoquery"]:
+		sourcelist = source["title"]["Source"].split(":::")
```
and you'll get a *much* nicer output format from `mwrogue` too

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,54 @@
+				else:
+					print("Error with tweet! {0} - {1}".format(str(r.errors[0]["title"]), str(source[0])))
+
+	print("Done!")
```
leave out any printing successes in the final version cos it'll run in crontab so we only want it to "make noise" when there's failures

arbolitoloco1 committed:
Update check_tweets.py

Now using mwrogue, not printing to console, and client updated to twitter_client. I still have doubts with how to handle ratelimit

arbolitoloco1 committed:
Update check_tweets.py

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,57 @@
+					r = twitter_client.get_tweet(splitStatus)
+				if not r.errors:
+					continue
+				elif r.errors[0]["title"] == "Not Found Error":
```
you don't need an `elif` after a `continue`, you can just de-indent

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,57 @@
+				continue
+			source = source.split(";;;")
+			if source[2] == "twitter.com":
+				splitStatus = re.search(r"status/([0-9]+)", source[0])[1]
```
can you call `splitStatus` something like `tweet_id` (I think that's what it is right?) or if that's not what it is, something else that's more descriptive?

RheingoldRiver commented on check_tweets.py:
```diff
@@ -0,0 +1,57 @@
+		for source in sourcelist:
+			if not source:
+				continue
+			source = source.split(";;;")
```
you use source[0] enough later on that I'd give it a name here

RheingoldRiver reviewed (commented):
Just a couple small style things really, otherwise looks great!

arbolitoloco1 committed:
Update check_tweets.py
